pkg/daemon/ceph/osd/volume.go

```diff
@@ -1161,7 +1161,17 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 			if err != nil {
 				return nil, errors.Wrapf(err, "failed to get device info for %q", blockPath)
 			}
-			osd.DeviceClass = sys.GetDiskDeviceClass(diskInfo)
```

Comment:
Since moving this back inside the `skipDeviceClass` code block, did you retest? I believe it's correct, but just wanted to know if your testing still looks good.

Reply:
I suggest creating a new function to set `osd.DeviceClass` and write tests for this new function.

Reply:
@travisn yes I tested after changes, the device class label is set correctly